### PR TITLE
New approach envoy v3- part 3

### PR DESCRIPTION
**Pull request by arianaw66**

### CODE REVIEW

**Comment by arianaw66** [comment]
Passing CI and GitHub build :)

**Comment by arianaw66** [comment]
/kick
known flake: #3226
build logs: https://console.cloud.google.com/cloud-build/builds/9f8621ae-b4db-422c-8839-ff46244265e2;step=14?project=solo-public

File: projects/gloo/pkg/plugins/gzip/plugin.go [hunk 1]
```diff
@@ -33,10 +37,58 @@ func (p *Plugin) HttpFilters(_ plugins.Params, listener *v1.HttpListener) ([]plu
 		return nil, nil
 	}
 
-	gzipFilter, err := plugins.NewStagedFilterWithConfig(wellknown.Gzip, gzipConfig, pluginStage)
+	envoyGzipConfig, err := convertGzip(gzipConfig) // Note: this modifies the gzipConfig
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/gzip/plugin.go:40]: lets make this a bit clearer that we are converting from v2 to v3 envoy gzip APIs here (e.g., rename `convertGzip`)

File: projects/gloo/pkg/plugins/gzip/plugin_test.go [hunk 2]
```diff
@@ -42,10 +127,12 @@ var _ = Describe("Plugin", func() {
 					ConfigType: &envoyhcm.HttpFilter_TypedConfig{
 						TypedConfig: utils.MustMessageToAny(&envoygzip.Gzip{
 							MemoryLevel:         &wrappers.UInt32Value{Value: 10.000000},
-							ContentLength:       &wrappers.UInt32Value{Value: 10.000000},
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/gzip/plugin_test.go:45]: why this change?

File: projects/gloo/pkg/plugins/gzip/plugin.go (hunk 1 above)
Comment (arianaw66) [reply projects/gloo/pkg/plugins/gzip/plugin.go:40]: I would argue that we're converting from glooGzip to envoyGzip, and that that's a more useful naming than which version of Envoy. Okay if I rename `convertGzip` to `glooToEnvoyGzip` or similar?

File: projects/gloo/pkg/plugins/gzip/plugin_test.go (hunk 2 above)
Comment (arianaw66) [reply projects/gloo/pkg/plugins/gzip/plugin_test.go:45]: Was just separating out test cases including and not including deprecated fields.

File: projects/gloo/pkg/utils/ssl.go [hunk 3]
```diff
@@ -335,3 +335,14 @@ func convertVersion(v v1.SslParameters_ProtocolVersion) (envoyauth.TlsParameters
 
 	return envoyauth.TlsParameters_TLS_AUTO, TlsVersionNotFoundError(v)
 }
+
+func convertSanList(sanList []string) []*envoymatcher.StringMatcher {
```
Comment (kdorosh) [inline projects/gloo/pkg/utils/ssl.go:339]: rename to `sanListToStringMatcherList`? or something more descriptive?

File: projects/gloo/pkg/plugins/gzip/plugin.go (hunk 1 above)
Comment (kdorosh) [reply projects/gloo/pkg/plugins/gzip/plugin.go:40]: agreed that is a better name 👍

File: projects/gloo/pkg/utils/ssl.go (hunk 3 above)
Comment (arianaw66) [reply projects/gloo/pkg/utils/ssl.go:339]: Was just matching convention in that file, but sure

File: projects/gloo/pkg/plugins/extauth/filter.go [hunk 4]
```diff
@@ -6,10 +6,10 @@ import (
 
 	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
 
-	envoycore "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
-	envoyauth "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/ext_authz/v2"
-	envoytype "github.com/envoyproxy/go-control-plane/envoy/type"
-	envoymatcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
+	envoycore "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoyauth "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/ext_authz/v3"
```
Comment (EItanya) [inline projects/gloo/pkg/plugins/extauth/filter.go:10]: @yuval-k is it ok to upgrade this filter now, unsure if `extauth v3` transport is safe yet, and how to enable/disable it. Can't seem to find it in the docs.

File: projects/gloo/pkg/plugins/gzip/plugin.go [hunk 5]
```diff
@@ -33,10 +37,58 @@ func (p *Plugin) HttpFilters(_ plugins.Params, listener *v1.HttpListener) ([]plu
...
+	// convert type from v2.Gzip to envoygzip.Gzip using json as an intermediary
+	jase := jsonpb.Marshaler{}
+	gzipStr, err := jase.MarshalToString(gzip)
+	if err != nil {
+		return nil, err
+	}
+	envoyGzip := new(envoygzip.Gzip)
+	if err := jsonpb.UnmarshalString(gzipStr, envoyGzip); err != nil {
+		return nil, err
+	}
```
Comment (EItanya) [inline projects/gloo/pkg/plugins/gzip/plugin.go:79]: I am against using json as an intermediary, would rather be explicit and copy over all fields like we do elsewhere. Much better to catch these errors at compile time.

File: projects/gloo/pkg/plugins/ratelimit/raw_util.go [hunk 6]
```diff
@@ -5,7 +5,7 @@ import (
 	"time"
 
 	envoyvhostratelimit "github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
-	envoyratelimit "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/rate_limit/v2"
+	envoyratelimit "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/ratelimit/v3"
```
Comment (EItanya) [inline projects/gloo/pkg/plugins/ratelimit/raw_util.go:8]: @yuval-k same question here as with extauth

File: projects/gloo/pkg/plugins/extauth/filter.go (hunk 4 above)
Comment (yuval-k) [reply projects/gloo/pkg/plugins/extauth/filter.go:10]: the filter will default to v2 transport, so this should be fine.
ideally our extauth/ratelimit server should support both v2 and v3 to allow a transition, but that's a separate issue

Comment (EItanya) [reply projects/gloo/pkg/plugins/extauth/filter.go:10]: How do you switch the transport version, I don’t even see the option

Comment (marcogschmidt) [reply projects/gloo/pkg/plugins/extauth/filter.go:10]: @EItanya I think it's the `transport_api_version` [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/ratelimit/v3/rls.proto#config-ratelimit-v3-ratelimitserviceconfig) and [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto.html?highlight=ext_authz#extensions-filters-http-ext-authz-v3-extauthz).

Comment (yuval-k) [reply projects/gloo/pkg/plugins/extauth/filter.go:10]: that's right, it's pretty new; do note that envoy always sends the v3 of the proto; the only reason it works right now is because they are binary compatible; either way shouldn't be a problem for this PR

**Comment by arianaw66** [comment]
/kick
known flake: #3226
build logs: https://console.cloud.google.com/cloud-build/builds/595a1c3b-3584-40db-858a-623b369cb5fb;step=14?project=solo-public

**Comment by arianaw66** [comment]
/kick
known flake: #3226
build logs: https://console.cloud.google.com/cloud-build/builds/f10b2afb-ac42-431b-94af-3a6bebcea13d;step=14?project=solo-public

**Comment by arianaw66** [comment]
/kick
`glooctl check` timeout flake
https://console.cloud.google.com/cloud-build/builds/c6d9bd1a-2d10-4f4c-b960-b0839d58c828;step=14?project=solo-public
This is annoying

**Comment by arianaw66** [comment]
https://console.cloud.google.com/cloud-build/builds/c309bd27-3b29-4272-81ea-f006ebef26a5;step=14?project=solo-public
